pkg/reconciler/ingress/ingress.go

```diff
@@ -503,6 +504,30 @@ func shouldReconcileTLS(ing *v1alpha1.Ingress) bool {
 	return isIngressPublic(ing) && len(ing.Spec.TLS) > 0
 }
 
+func shouldReconcileHTTPServer(ctx context.Context, ing *v1alpha1.Ingress) bool {
+	// We will create a Ingress specific HTTPServer when
+	// 1. auto TLS is enabled as in this case users want us to fully handle the TLS/HTTP behavior,
+	// 2. HTTPOption is set to Redirected as we don't have default HTTP server supporting HTTP redirection.
+	return isIngressPublic(ing) && (ing.Spec.HTTPOption == v1alpha1.HTTPOptionRedirected || autoTLSEnabled(ctx, ing))
```

Comment:
```suggestion
	return isIngressPublic(ing) && (ing.Spec.HTTPOption == v1alpha1.HTTPOptionRedirected || len(ing.Spec.TLS) > 0)
```

I think it should be consistent with `shouldReconcileTLS()`. I assume that `len(ing.Spec.TLS) > 0)` means autoTLS is enabled but please correct me if I was wrong.

Reply:
Yea they are almost the same right now. I changed it per your suggestion.